tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -197,6 +271,27 @@ private void ConfigureTraceTransport(IConfigurationSource source, bool isWindows
 #endif
             }
 
+            if (traceTransport == null)
+            {
+                // Check for UDS
+                var traceSocket = source?.GetString(ConfigurationKeys.TracesUnixDomainSocketPath);
+
+                if (traceSocket != null)
+                {
+                    traceTransport = TracesTransportType.UnixDomainSocket;
+                    TracesUnixDomainSocketPath = traceSocket;
+                }
```

Comment:
So we have _two_ ways to set UnixDomainSocketPath? via `agentHost` _or_ `ConfigurationKeys.TracesUnixDomainSocketPath`? Do we need the latter one at all?

Reply:
There is some uncomfortable ambiguity for me here as well.
I loosely followed java's examples (https://github.com/DataDog/dd-trace-java/pull/2666/files#diff-f5198182d1861fa2cdb96cfd23e3fd1efc51e3727e9c8e080b777bdccaa2b7ccR114)
and had some slack conversations about detecting UDS through the host variable.

Consider also that the "agent host" is not necessarily the "trace agent host" or the "dogstatsd host"...
The agent host being the source of truth really only makes sense in the case of default TCP/UDP.

There are historical variables that call out trace agent host, which is what we probably should be checking against for the UDS prefix, but again... Was complicating this config worth handling all of these ambiguous cases which would otherwise be solved by relying on convention or being explicit in setting the main variables?

I don't know. I don't really like that "agent host" can be used to set UDS... But it was a thing in the other tracers.

It's not technically in the RFC... It doesn't *BOTHER* me enough to care to remove it.
But, we don't have anyone relying on it... It was maybe a backward compatibility thing for Java? 

I'll ask them.

Reply:
It looks like `ConfigurationKeys.TracesUnixDomainSocketPath == DD_APM_RECEIVER_SOCKET` is used for trace agent configuration (see https://github.com/DataDog/datadog-agent/blob/c06b6c8663747467dad0bbdcb00f6474c806aae2/pkg/config/apm.go#L79) so we overload it in the Tracer for convenience. Could we use only this environment variable and not check `agentHost` when considering Unix Domain Socket configuration?

Reply:
I think we could only use the variable yes, I was following suit with java, which maybe doesn't make sense.